### PR TITLE
Revise “Container runtimes” page in light of dockershim removal

### PR DESCRIPTION
**Pull request by sftim**
This updates the [Container Runtimes](https://kubernetes.io/docs/setup/production-environment/container-runtimes/) page of the website to include information about the dockershim deprecation, so that the users can take an informed decision. This helps with issue #28449 and potentially supersedes PR #30882.

### CODE REVIEW

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 1]
```diff
@@ -13,29 +13,43 @@ You need to install a
...
+This page provides an outline of how to use several common container runtimes with
+Kubernetes.
 
 - [containerd](#containerd)
 - [CRI-O](#cri-o)
 - [Docker Engine](#docker)
 - [Mirantis Container Runtime](#mcr)
 
 {{< note >}}
-For other operating systems, look for documentation specific to your platform.
```
Comment (tengqm) [inline content/en/docs/setup/production-environment/container-runtimes.md:33]: The original content here at least mentions other operating systems. Where this information is relocated?

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 2]
```diff
@@ -120,95 +134,51 @@ using the (deprecated) v1alpha2 API instead.
...
-   Copy-Item -Path ".\bin\" -Destination "$Env:ProgramFiles\containerd" -Recurse -Force
-   cd $Env:ProgramFiles\containerd\
-   .\containerd.exe config default | Out-File config.toml -Encoding ascii
-
-   # Review the configuration. Depending on setup you may want to adjust:
-   # - the sandbox_image (Kubernetes pause image)
-   # - cni bin_dir and conf_dir locations
-   Get-Content config.toml
-
-   # (Optional - but highly recommended) Exclude containerd from Windows Defender Scans
```
Comment (tengqm) [inline content/en/docs/setup/production-environment/container-runtimes.md:197]: Why drop these lines?

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 3]
```diff
@@ -232,205 +202,43 @@ When using kubeadm, manually configure the
...
-
+To install CRI-O, follow [CRI-O Install Instructions](https://github.com/cri-o/cri-o/blob/main/install.md#readme).
 
 #### cgroup driver
 
-CRI-O uses the systemd cgroup driver per default. To switch to the `cgroupfs`
-cgroup driver, either edit `/etc/crio/crio.conf` or place a drop-in
-configuration in `/etc/crio/crio.conf.d/02-cgroup-manager.conf`, for example:
+CRI-O uses the systemd cgroup driver per default, which is likely to work fine
+for you. To switch to the `cgroupfs` cgroup driver, either editi
```
Comment (tengqm) [inline content/en/docs/setup/production-environment/container-runtimes.md:210]:
```suggestion
for you. To switch to the `cgroupfs` cgroup driver, either edit
```

File: content/en/docs/setup/production-environment/container-runtimes.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/setup/production-environment/container-runtimes.md:33]: Do you mean OSs other than Linux or Windows?

My thinking is that, given our docs policy on 3rd party content, the container runtimes' docs should cover installs onto various operating systems. If they don't, folks can propose adding it to _those_ docs, not to Kubernetes.

File: content/en/docs/setup/production-environment/container-runtimes.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/setup/production-environment/container-runtimes.md:197]: https://kubernetes.io/docs/contribute/style/content-guide/#what-s-allowed basically. We should aim to link to external docs; I'm also happy to see work done on those external docs (and maybe I will do some of that myself).

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-32738--kubernetes-io-main-staging.netlify.app/docs/setup/production-environment/container-runtimes/#docker

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 4]
```diff
@@ -13,29 +13,43 @@ You need to install a
...
+[Check whether Dockershim deprecation affects you](/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-deprecation-affects-you/) to understand how this removal might
+affect you. To learn about migrating from using dockershim, see
+[Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/).
+
+If you are running a version of Kubernetes other than v{{< skew currentVersion >}},
+check the documentation for that version.
 {{< /note >}}
 
+
+<!-- body -->
```
Comment (PI-Victor) [inline content/en/docs/setup/production-environment/container-runtimes.md:47]: what is this comment doing here?

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 5]
```diff
@@ -120,95 +134,51 @@ using the (deprecated) v1alpha2 API instead.
 
 ### containerd
 
-This section contains the necessary steps to use containerd as CRI runtime.
+This section outlines the necessary steps to use containerd as CRI runtime.
```
Comment (PI-Victor) [inline content/en/docs/setup/production-environment/container-runtimes.md:137]: should it be `containerd`?

**Review by PI-Victor — COMMENTED** [review]
small nits, otherwise looks good from my end

**Comment by kbhawkey** [comment]
@sftim, are the changes part of the Dockershim removal updates?

**Comment by sftim** [comment]
> are the changes part of the Dockershim removal updates?

Yes, these changes are related to the dockershim removal.

File: content/en/docs/setup/production-environment/container-runtimes.md (hunk 5 above)
Comment (sftim) [reply content/en/docs/setup/production-environment/container-runtimes.md:137]: We don't need code style for the tool name (which is containerd).

**Comment by kbhawkey** [comment]
/lgtm

Possibly move to `dev-1.24` branch?

**Comment by sftim** [comment]
These changes document the story for v1.23

I expect that the existing v1.24 version of this page would “win” the merge conflict. The changes are important because people are still using v1.23 and may want to move away from dockershim already, even though it is only deprecated.

**Pull request merged**